### PR TITLE
update the gpu-aware-mpi page

### PR DESCRIPTION
**Pull request by edgargabriel**
Three changes:
 - add the ucx compatibility table
 - add the --with-rocm=/opt/rocm option to the compilation of Open MPI
 - add a section about how to compile and use UCC for collective operations.

### CODE REVIEW

File: docs/how_to/gpu_aware_mpi.md [hunk 1]
```diff
@@ -75,6 +75,13 @@ make -j $(nproc)
 make -j $(nproc) install
 ```
 
+The following table shows the compatibility of UCX versions with ROCm versions.
+
```
Comment (saadrahim) [inline docs/how_to/gpu_aware_mpi.md:79]: Can a link to the compatibility information be placed here instead?

Comment (edgargabriel) [reply docs/how_to/gpu_aware_mpi.md:79]: Actually, the information is at the moment only available in a Confluence page, so I am not sure where to add it other wise

Comment (saadrahim) [reply docs/how_to/gpu_aware_mpi.md:79]: https://rocmdocs.amd.com/en/develop/release/3rd_party_support_matrix.html#communication-libraries

Comment (saadrahim) [reply docs/how_to/gpu_aware_mpi.md:79]: The information is already out.

Comment (edgargabriel) [reply docs/how_to/gpu_aware_mpi.md:81]: done

Comment (saadrahim) [reply docs/how_to/gpu_aware_mpi.md:81]: I will change the link to relative.

**Review by saadrahim — APPROVED** [review]

**Pull request merged**